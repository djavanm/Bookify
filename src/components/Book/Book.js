import React from 'react';

const Book = ({data, toggleFavorite, isFavorite}) => {
  const { author_name, book_name, artwork_url, description, release_date, book_id } = data;
  return (
    <article className='book'>
      <img src={artwork_url} alt={`Cover for ${book_name}`} />
      <p>{author_name}</p>
      <p>{book_name}</p>
      <button onClick={() => toggleFavorite(data, isFavorite)}>Fav</button>
    </article>
  )
}

export default Book;
