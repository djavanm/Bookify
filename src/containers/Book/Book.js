import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';


export const Book = ({currentUser, data, toggleFavorite, favorites}) => {
  const { author_name, book_name, artwork_url, book_id } = data;
  let shortName = book_name.split(' ')
  if(shortName.length > 5) {
    shortName = [shortName[0], shortName[1],shortName[2], shortName[3], shortName[4], shortName[5], shortName[6], ' ...']
  }
  shortName = shortName.join(' ')
  const isFavorite = favorites.map(favorite => favorite.book_id).includes(book_id);
  const btn = isFavorite ? 'btn-active': 'btn'
  const favoriteBtn = currentUser ? <button className={btn} onClick={() => toggleFavorite(data, isFavorite)}>Fav</button> : <Link to='/login'><button>Fav</button></Link>
  return (
    <article className='book'>
      <img src={artwork_url} alt={`Cover for ${book_name}`} className='card-cover' />
      <section className='card-info'>
      <Link to={`/book/${book_id}`} className="link">
        <p className='book-name'>{shortName}</p>
      </Link>
      <p className='author-name'>{author_name}</p>
      </section>
      {favoriteBtn}
    </article>
  )
}

export const mapStateToProps = state => ({
  currentUser: state.currentUser,
  favorites: state.favorites
});

export default connect(mapStateToProps, null)(Book);
