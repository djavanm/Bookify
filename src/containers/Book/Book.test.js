import React from 'react';
import { shallow } from 'enzyme';
import { Book } from './Book';

describe('Book', () => {
  let wrapper
  const currentUserMock = {
    name: 'Sam',
    password: 'blah',
    email: 'blah@gmail.com'
  }
  const dataMock = {
      author_name: 'Neil Gaiman',
      book_name: 'Good Omens',
      artwork_url: 'https://images/good',
      book_id: 42
    }
  const toggleFavoriteMock = jest.fn();
  const favoritesMock = []

  beforeEach(() => {
    wrapper = shallow(<Book
      currentUser={currentUserMock}
      data={dataMock}
      toggleFavorite={toggleFavoriteMock}
      favorites={favoritesMock}
      />)
  })

  it('should', () => {
    expect(wrapper).toMatchSnapshot();
  })
})