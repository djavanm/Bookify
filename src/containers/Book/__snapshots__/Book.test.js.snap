// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book should 1`] = `
<article
  className="book"
>
  <img
    alt="Cover for Good Omens"
    className="card-cover"
    src="https://images/good"
  />
  <section
    className="card-info"
  >
    <Link
      className="link"
      to="/book/42"
    >
      <p
        className="book-name"
      >
        Good Omens
      </p>
    </Link>
    <p
      className="author-name"
    >
      Neil Gaiman
    </p>
  </section>
  <button
    className="btn"
    onClick={[Function]}
  >
    Fav
  </button>
</article>
`;
